source/val/validation_state.h

```diff
@@ -359,8 +369,17 @@ class ValidationState_t {
   SpvAddressingModel addressing_model_;
   SpvMemoryModel memory_model_;
 
+  // Is OpTypeInt allowed with 16 bit width?
```

Comment:
`permit_int16_type_` and `permit_float16_type_` variables are not used. I think the `Feature` members are meant to do the work instead?